src/lib.rs

```diff
@@ -14,6 +14,10 @@
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
 #![deny(unused_must_use, rust_2018_idioms)]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc as std;
```

Comment:
Can you remove the alias? I prefer being explicit about `alloc`, plus it doesn't re-export items from `core` like `std` does.